test/run_tests.sh

```diff
@@ -2,114 +2,139 @@
-PATCH_TEST_FILE=$(find test/patch/*_test.rb -type f)
-for file in $PATCH_TEST_FILE; do
-  check_file_name=$file
-  BUNDLE_GEMFILE=gemfiles/test_gems.gemfile DBO_PATCH_TEST=1 bundle exec ruby -I test $file
-  check_status
-done
+    check_test_result "$relative_path" "$output" "$status"
+}
 
-# for resource detection test
-RESOURCE_DETECTOR_TEST_FILE=$(find test/support/resource_detector/aws/*_test.rb -type f)
-for file in $RESOURCE_DETECTOR_TEST_FILE; do
-  check_file_name=$file
-  BUNDLE_GEMFILE=gemfiles/test_gems.gemfile RESOURCE_TEST=1 bundle exec ruby -I test $file
-  check_status
+# Setup logging
+time=$(date "+%Y%m%d_%H%M")
+export TEST_RUNS_FILE_NAME="./log/testrun_$time.log"
+
+# Remove previous log files
+rm -f ./log/*.log
+
+log_message "=== SolarWinds APM Ruby Test Runner ==="
+log_message "Test pattern: $test_pattern"
+log_message "Ruby version: $(ruby -e 'print(RUBY_VERSION)'); $(bundle -v)"
+log_message "Searching for test files matching: test/**/$test_pattern"
+test_files=$(find test -name "$test_pattern" -type f | sort)
+
+# Count total files to run
+total_files=$(echo "$test_files" | wc -l | tr -d ' ')
+log_message "Found $total_files test files to run"
+
+# Run each test file
+# test_files=(test/solarwinds_apm/init_test/init_1_test.rb)
+current_file=0
+for test_file in $test_files; do
+    current_file=$((current_file + 1))
+    log_message "[$current_file/$total_files] Processing: $test_file"
+    run_test_file "$test_file"
 done
 
-# for otlp processor test
-OTLP_PROCESSOR_TEST_FILE=$(find test/opentelemetry/otlp_processor_*_test.rb -type f)
-for file in $OTLP_PROCESSOR_TEST_FILE; do
-  check_file_name=$file
-  BUNDLE_GEMFILE=gemfiles/test_gems.gemfile bundle exec ruby -I test $file
-  check_status
-done
+# Generate summary
+log_message ""
+log_message "=== TEST EXECUTION SUMMARY ==="
+log_message "Total files processed: $total_tests"
+log_message "Passed: $passed_tests"
+log_message "Failed: $failed_tests"
+log_message "Success rate: $((passed_tests * 100 / total_tests))%"
+
+if [[ ${#passed_files[@]} -gt 0 ]]; then
+    log_message ""
+    log_message "✅ PASSED FILES:"
+    for file in "${passed_files[@]}"; do
+        log_message "  - $file"
+    done
+fi
 
-NUMBER_FILE=$(find test/solarwinds_apm/init_test/*_test.rb -type f | wc -l)
-for ((i = 1; i <= $NUMBER_FILE; i++)); do
-  check_file_name=init_${i}_test.rb
-  BUNDLE_GEMFILE=gemfiles/test_gems.gemfile bundle exec ruby -I test test/solarwinds_apm/init_test/init_${i}_test.rb
-  check_status
-done
+if [[ ${#failed_files[@]} -gt 0 ]]; then
+    log_message ""
+    log_message "❌ FAILED FILES:"
+    for file in "${failed_files[@]}"; do
+        log_message "  - $file"
+    done
+fi
 
+# Print summary to console
 echo ""
-echo "--- SUMMARY ------------------------------"
-grep -E '===|failures|FAIL|ERROR' "$TEST_RUNS_FILE_NAME"
+echo "==================== FINAL SUMMARY ===================="
+echo "Total: $total_tests | Passed: $passed_tests | Failed: $failed_tests"
```

Comment:
```suggestion
echo "Total Files: $total_tests | Passed: $passed_tests | Failed: $failed_tests"
```
I _think_ this "total_tests" is actually tracking the total test files (not actual test cases) that were run.